### ISSUE TITLE
Learn Retrofit Library

### ISSUE THREAD

**Issue by YigitSekerci** [bounswe/bounswe2023group4#218]
### Problem Definition
Retrofit is network request library for android apps and every member should learn how to use it in order to send backend requests. 

### Acceptance Criteria
* Learn what does retrofit library do
* Learn how to implement retrofit library

### Suggested Solutions
* Retrofit is very well known library team members can find many examples in both android guides and youtube.

**Comment by aemresafak** [comment]
I have studied Retrofit and learned that it is a type-safe HTTP client for Android, Java, and Kotlin that makes it easier to consume RESTful web services. It provides a powerful framework for authenticating and interacting with APIs, and sending network requests with OkHttp.

I have also learned how to implement Retrofit in Android. This includes adding the necessary dependencies to my project, creating a Retrofit interface, defining POJOs to represent the JSON responses, and making network requests.

I have found many examples of using Retrofit in Android guides and on YouTube. These examples have been helpful in learning how to use Retrofit in my own projects.

Overall, I am confident that I can use Retrofit to send backend requests in my Android apps.

**Comment by YigitSekerci** [comment]
I have learned basics of retrofit library and implemented an example service in our project to try this concept. I have learned how does converters in response works. Since retrofit have many config I couldn't cover them all but I guess I have learned enough concepts to start calling API from mobile apps.

**Comment by AliAlperenSonmez** [comment]
I've gained a comprehensive understanding of Retrofit as a network request library, recognizing its essential role in facilitating backend communication in Android apps. I am proficient in implementing Retrofit within Android projects, demonstrating my ability to efficiently utilize this library to send and manage network requests. I have explored various resources, including Android guides and YouTube tutorials, to learn and apply Retrofit effectively.
